### ISSUE TITLE
SystemError: initialization of _internal failed without raising an exception when importing causalpy in jupyter notebook

### ISSUE THREAD

**Issue by dushyant-mishra** [pymc-labs/CausalPy#171]
when I import the package with "import causalpy as cp" command in jupyter notebook I get the following error:
SystemError: initialization of _internal failed without raising an exception. See following:

SystemError                               Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_56240/1409921450.py in <module>
----> 1 import causalpy as cp

~\anaconda3\lib\site-packages\causalpy\__init__.py in <module>
----> 1 import arviz as az
      2 
      3 import causalpy.pymc_experiments
      4 import causalpy.pymc_models
      5 import causalpy.skl_experiments

~\anaconda3\lib\site-packages\arviz\__init__.py in <module>
     32 
     33 from .data import *
---> 34 from .plots import *
     35 from .plots.backends import *
     36 from .stats import *

~\anaconda3\lib\site-packages\arviz\plots\__init__.py in <module>
      1 """Plotting functions."""
----> 2 from .autocorrplot import plot_autocorr
      3 from .bpvplot import plot_bpv
      4 from .bfplot import plot_bf
      5 from .compareplot import plot_compare

~\anaconda3\lib\site-packages\arviz\plots\autocorrplot.py in <module>
      5 from ..rcparams import rcParams
      6 from ..utils import _var_names
----> 7 from .plot_utils import default_grid, filter_plotters_list, get_plotting_function
      8 
      9 

~\anaconda3\lib\site-packages\arviz\plots\plot_utils.py in <module>
     14 from ..rcparams import rcParams
     15 from ..stats.density_utils import kde
---> 16 from ..stats import hdi
     17 
     18 KwargSpec = Dict[str, Any]

~\anaconda3\lib\site-packages\arviz\stats\__init__.py in <module>
      2 """Statistical tests and diagnostics for ArviZ."""
      3 from .density_utils import *
----> 4 from .diagnostics import *
      5 from .stats import *
      6 from .stats import _calculate_ics

~\anaconda3\lib\site-packages\arviz\stats\diagnostics.py in <module>
     13 from ..utils import Numba, _numba_var, _stack, _var_names
     14 from .density_utils import histogram as _histogram
---> 15 from .stats_utils import _circular_standard_deviation, _sqrt
     16 from .stats_utils import autocov as _autocov
     17 from .stats_utils import not_valid as _not_valid

~\anaconda3\lib\site-packages\arviz\stats\stats_utils.py in <module>
    535 
    536 @conditional_vect
--> 537 def _sqrt(a_a, b_b):
    538     return (a_a + b_b) ** 0.5
    539 

~\anaconda3\lib\site-packages\arviz\utils.py in conditional_vect(function, **kwargs)
    303 
    304     if function:
--> 305         return wrapper(function)
    306     else:
    307         return wrapper

~\anaconda3\lib\site-packages\arviz\utils.py in wrapper(function)
    296     def wrapper(function):
    297         try:
--> 298             numba = importlib.import_module("numba")
    299             return numba.vectorize(**kwargs)(function)
    300 

~\anaconda3\lib\importlib\__init__.py in import_module(name, package)
    125                 break
    126             level += 1
--> 127     return _bootstrap._gcd_import(name[level:], package, level)
    128 
    129 

~\anaconda3\lib\site-packages\numba\__init__.py in <module>
     41 
     42 # Re-export vectorize decorators and the thread layer querying function
---> 43 from numba.np.ufunc import (vectorize, guvectorize, threading_layer,
     44                             get_num_threads, set_num_threads)
     45 

~\anaconda3\lib\site-packages\numba\np\ufunc\__init__.py in <module>
      1 # -*- coding: utf-8 -*-
      2 
----> 3 from numba.np.ufunc.decorators import Vectorize, GUVectorize, vectorize, guvectorize
      4 from numba.np.ufunc._internal import PyUFunc_None, PyUFunc_Zero, PyUFunc_One
      5 from numba.np.ufunc import _internal, array_exprs

~\anaconda3\lib\site-packages\numba\np\ufunc\decorators.py in <module>
      1 import inspect
      2 
----> 3 from numba.np.ufunc import _internal
      4 from numba.np.ufunc.parallel import ParallelUFuncBuilder, ParallelGUFuncBuilder
      5 

SystemError: initialization of _internal failed without raising an exception

**Comment by drbenvincent** [comment]
Thanks for reporting this. It looks like something to do with arviz and/or numba. This is a bit odd, so it's making me suspect an issue to do with the environment.

And can I check, you pip installed causalpy into an environment and you have that environment active in your Jupyter lab session?

And let me double check: jupyter or jupyterlab? I'll see if I can reproduce the problem.

Are you running on an Apple Silicon machine. Sometimes there are issues with that an PyMC, but to be honest it doesn't look like that's the issue.

**Comment by dushyant-mishra** [comment]
"And can I check, you pip installed causalpy into an environment and you have that environment active in your Jupyter lab session?" Yes I did a pip install as mentioned here: https://causalpy.readthedocs.io/en/latest/

"And let me double check: jupyter or jupyterlab? I'll see if I can reproduce the problem." I am on jupyter notebook

"Are you running on an Apple Silicon machine. Sometimes there are issues with that an PyMC, but to be honest it doesn't look like that's the issue." I have a windows 11 intel machine running jupyter notebook

Sorry I am new to this and was checking out the cool package you all have built, BTW congratulations on that!

**Comment by drbenvincent** [comment]
I'll take a look. Apologies if it takes a day or two, I'm coming down with a cold. In the mean time it would be interesting to see if it works in a jupyterlab notebook.

**Comment by drbenvincent** [comment]
I created (and activated) an environment called `cptest_env` with: 
```
mamba create -n cptest_env python=3.10 notebook nb_conda causalpy
mamba activate cptest_env
```

Booted up a notebook with:
```
jupyter notebook
```
Then:
* opened a new notebook with the right environment (`cptest_env`)
* I was able to `import causalpy as cp` with no problem
* I was then able to paste in the rest of the QuickStart [here](https://causalpy.readthedocs.io/en/latest)

It all worked fine for me. So I don't think the issue is `jupyter notebook` (which I'd not tried before) or `jupyterlab` (which I'd always used in the past).

So I think it's something to do with the environment, or something PC related. I don't have a PC to test this with I'm afraid. So if following these instructions doesn't work then we'll have to see if we can find someone with a PC who might be able to replicate the problem.
